src/system/process.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import subprocess
+import tempfile
+
+import psutil  # type: ignore
+
+
+class Process:
+    def __init__(
+        self
+    ):
+        self.process = None
+        self.stdout = None
+        self.stderr = None
+        self.has_started = False
+
+    def start(self, command, cwd):
+        if self.has_started:
+            raise AssertionError("Cannot start a started process!")
```

Comment:
Add a class called `ProcessStartedError` and make its message to include the PID of the already started process.

The convention for messages is not to include any kind of exclamations, so something like `f"Process already started, pid: {pid}"`